File: Day05/part1.py
with open("input", "r") as f:
    ranges = [[0] * 2] * 190
    lines = f.readlines()
    for index in range(189):
        ranges[index] = list(map(int, lines[index].split("-")))
    counter = 0
    for index in range(191, len(lines)):
        for i in range(len(ranges)):
            if int(lines[index][:-1]) in range(ranges[i][0] + 1, ranges[i][1]):
                counter += 1
                break
    print(f"The answer is: {counter}.")

File: Day05/part1.c
#include <stdint.h>
#include <stdio.h>
#include <stdlib.h>

int main(void) {
    FILE *f = fopen("input", "r");
    if (f == NULL) return 1;
    uint64_t ranges[190][2] = {0};
    char buffer[100];
    int i = 0;
    while (i < 190) {
        fgets(buffer, 100, f);
        sscanf(buffer, "%lu-%lu", &ranges[i][0], &ranges[i][1]);
        i++;
    }
    fgets(buffer, 100, f);
    int counter = 0;
    while (fgets(buffer, 100, f) != NULL) {
        uint64_t num = strtoull(buffer, NULL, 10);
        for (int j = 0; j < 190; j++) {
            if (ranges[j][0] < num && num < ranges[j][1]) {
                counter++;
                break;
            }
        }
    }
    printf("The answer is: %d.", counter);
    return 0;
}
